gateway/ln-gateway/src/lib.rs

```diff
@@ -1636,30 +1636,20 @@ impl Gateway {
 
             for (federation_id, federation_config) in federations {
                 if let Some(client) = self.federation_manager.read().await.client(federation_id) {
-                    if async {
-                        client
-                            .value()
-                            .get_first_module::<GatewayClientModule>()?
-                            .register_with_federation(
-                                route_hints.clone(),
-                                GW_ANNOUNCEMENT_TTL,
-                                federation_config.fees,
-                                lightning_context.clone(),
-                            )
-                            .await
-                    }
-                    .instrument(client.span())
-                    .await
-                    .is_err()
-                    {
-                        Err(AdminGatewayError::RegistrationError {
-                            federation_id: *federation_id,
-                        })?;
-                    }
+                    client
+                        .value()
+                        .get_first_module::<GatewayClientModule>()
+                        .expect("No GatewayClientModule exists")
+                        .register_with_federation(
+                            route_hints.clone(),
+                            GW_ANNOUNCEMENT_TTL,
+                            federation_config.fees,
+                            lightning_context.clone(),
+                        )
+                        .await;
```

Comment:
Registration should run in parallel, so that timeouts of earlier federations don't block later federations.

Reply:
Fixed. Cancellable task is now spawn for each registration in the loop.